.github/workflows/generate-docs.yml

```diff
@@ -0,0 +1,28 @@
+name: Deploy to GitHub Pages
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+# Based on https://github.com/marketplace/actions/deploy-to-github-pages#using-an-ssh-deploy-key-
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Build documentation in /docs
+        run: |
+          npm install
+          npm run docs
+
+      - name: Deploy /docs to branch gh-pages
```

Comment:
Are you sure you want to do it this way? Normally one uses _either_ the `docs/` directory _or_ the gh-pages branch, but not both. Do you have a reason for not using `docs/` on master?

Reply:
If we used `docs/` on master, won't we have to commit all the generated HTML files into the repository and then regenerate them every time we update the documentation? I think that would result in a lot of unnecessary commits and increase the repository size unnecessarily. If there's a way to avoid that, let me know -- that would certainly be simpler!

Note also that we're not serving the `docs/` directory from the gh-pages branch: this Github Action generates the documentation in the `docs/` directory, and then copies *only* the contents of the `docs/` directory to the [gh-pages branch](https://github.com/phyloref/phyx.js/tree/gh-pages). I think this is pretty standard for web applications hosted on Github Pages -- for instance, [Vue.js' manual](https://cli.vuejs.org/guide/deployment.html#github-pages)  recommends generating the website in the `dist/` directory and then publishing just that directory to the gh-pages branch. That's the method we currently use to deploy Klados.

Reply:
> If we used `docs/` on master, won't we have to commit all the generated HTML files into the repository and then regenerate them every time we update the documentation? I think that would result in a lot of unnecessary commits and increase the repository size unnecessarily.

The way I (and I think many others) handle this for an R project is to regenerate the web-based documentation only sporadically, for example when major changes are made, or before a new release is cut. Obviously if you run this automatically then presumably every merge to master will trigger this? It still needn't make the repo size much bigger if what changes aren't binary files.

> 
> Note also that we're not serving the `docs/` directory from the gh-pages branch: this Github Action generates the documentation in the `docs/` directory, and then copies _only_ the contents of the `docs/` directory to the [gh-pages branch](https://github.com/phyloref/phyx.js/tree/gh-pages).

So this wouldn't keep a commit history on gh-pages, instead every time the docs are regenerated, the gh-pages branch is reset?

Reply:
> The way I (and I think many others) handle this for an R project is to regenerate the web-based documentation only sporadically, for example when major changes are made, or before a new release is cut. Obviously if you run this automatically then presumably every merge to master will trigger this? It still needn't make the repo size much bigger if what changes aren't binary files.

Hmm. I like the idea of not having to do this manually every time we release -- note that although this Github Action has a `workflow_dispatch` trigger for testing, my intention is to rely on the documentation to be generated automatically every time we release a new version of Phyx.js (the `release: types: [published]` trigger). But you're right that once we generate the documentation, it shouldn't change too much between different versions. If you prefer that approach, it would look like PR #85. If you prefer that, feel free to approve it and close this PR.

> So this wouldn't keep a commit history on gh-pages, instead every time the docs are regenerated, the gh-pages branch is reset?

Nope, we [keep a history](https://github.com/phyloref/phyx.js/commits/gh-pages) of gh-pages branch changes as usual. It's not intended to be very useful, since it only covers files generated by this Github Action in the `docs/` directory, but it's there if needed.

Reply:
> > So this wouldn't keep a commit history on gh-pages, instead every time the docs are regenerated, the gh-pages branch is reset?
> 
> Nope, we [keep a history](https://github.com/phyloref/phyx.js/commits/gh-pages) of gh-pages branch changes as usual. It's not intended to be very useful, since it only covers files generated by this Github Action in the `docs/` directory, but it's there if needed.

Well, then you don't save repo size between `docs/` on master or `/` on gh-pages, right?

Reply:
It keeps it out of the `master` branch, which I think is nice. However, I'm happy to try this using the approach in PR #85 instead. I'll close this PR.